server/src/connectionHandlers/onDefinition.ts

```diff
@@ -132,3 +156,22 @@ function createDefinitionLocationForPathInfo (path: ParsedPath): Location {
 
   return location
 }
+
+function findFileInDirectory (dir: string, fileName: string): string | null {
+  try {
```

Comment:
Use a library for finding files, I don't think there is added value in defining this ourselves.
Use https://code.visualstudio.com/api/references/vscode-api#workspace.findFiles
 (or alternatively https://github.com/isaacs/node-glob)